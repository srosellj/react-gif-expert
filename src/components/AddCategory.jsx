import { useState } from "react"
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim()

        if (newInputValue.length <= 1) {
            return;
        }

        // setCategories(categories => [inputValue, ...categories])
        setInputValue('');
        onNewCategory(newInputValue)
    }
    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}
