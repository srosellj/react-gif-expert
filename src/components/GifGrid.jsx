import { GiftItem } from "./GiftItem"
import { useFetchGifts } from "../hooks/useFetchGifts"

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifts(category);

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando...</h2>)
            }

            <div className="card-gift">
                {
                    images.map((image) =>
                    (<GiftItem
                        key={image.id}
                        {...image}
                    />)
                    )
                }
            </div>
        </>
    )
}
