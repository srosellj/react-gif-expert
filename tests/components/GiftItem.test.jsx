import { render, screen } from "@testing-library/react"
import { GiftItem } from "../../src/components/GiftItem"

describe('Testeo de <GiftItem />', () => {

    const title = 'Saitama'
    const url = 'https://urlcualquera.com/saitama.jpg'

    test('Match del snapshot', () => {

        const { container } = render(<GiftItem title={title} url={url} />)

        expect(container).toMatchSnapshot();

    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GiftItem title={title} url={url} />)
        // screen.debug()
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);


    })

    test('Debe de mostrar el tÃ­tulo en el componente', () => {
        render(<GiftItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();

    })
})