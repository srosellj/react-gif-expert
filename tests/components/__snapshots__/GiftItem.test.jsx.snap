// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testeo de <GiftItem /> Match del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://urlcualquera.com/saitama.jpg"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
